kernelci/api/__init__.py

```diff
@@ -125,6 +125,36 @@ def create_node(self, node: dict) -> dict:
     def update_node(self, node: dict) -> dict:
         """Update an existing node object (with id)"""
 
+    # -----------
+    # User groups
+    # -----------
+
+    @abc.abstractmethod
+    def get_group(self, group_id: str) -> dict:
+        """Get the user group matching the given group id"""
+
+    @abc.abstractmethod
+    def get_groups(
+        self, attributes: dict,
+        offset: Optional[int] = None, limit: Optional[int] = None
+    ) -> Sequence[dict]:
+        """Get user groups that match the provided attributes"""
+
+    # -------------
+    # User profiles
+    # -------------
+
+    @abc.abstractmethod
+    def get_user_profile(self, user_profile_id: str) -> dict:
+        """Get the user profile matching the given profile id"""
+
+    @abc.abstractmethod
+    def get_user_profiles(
+        self, attributes: dict,
+        offset: Optional[int] = None, limit: Optional[int] = None
+    ) -> Sequence[dict]:
+        """Get user profiles that match the provided attributes"""
+
```

Comment:
I would suggest to leave this out for now as the API side of user profiles is not ready yet.